controllers/druid/handler.go

```diff
@@ -1308,7 +1324,8 @@ func makePodSpec(nodeSpec *v1alpha1.DruidNodeSpec, m *v1alpha1.Druid, nodeSpecUn
 		v1.Container{
 			Image:           firstNonEmptyStr(nodeSpec.Image, m.Spec.Image),
 			Name:            fmt.Sprintf("%s", nodeSpecUniqueStr),
-			Command:         []string{firstNonEmptyStr(m.Spec.StartScript, "bin/run-druid.sh"), nodeSpec.NodeType},
+			Command:         getCommand(nodeSpec, m),
+			Args:            getEntryArg(nodeSpec, m),
```

Comment:
instead of adding this complex custom logic, would it make more sense to let the user explicitly override `Command` and `Args` directly?

Reply:
In case of an override for a ```Command``` or an ```Arg``` how can it be done directly ? Override in the CR ? or some way of templating to override it out

Reply:
yes, I meant to let the user override those in the CR, rather than creating our own special variables

Reply:
Right now we have `startScript`, `druidScript`, and `entryArg`. The way these get combined is quite confusing, and it's not clear why we need 3.

If I specify `druidScript`, but neither `startScript` nor `entryArg`, then `druidScript` is ignored and we run the default `bin/run-druid.sh` so `druidScript` doesn't seem to control the Druid script in all cases unlike what intuition might suggest

This is quite confusing logic for someone new to this (especially since there is very little documentation).

Reply:
I tried to do that initially. The problem is that, we have to add the node type after the druid.sh script for each node dynamically, otherewise the user needs to specify separate container commands and arguments for each node. 
Therefore, we can not just let them overwrite the commands & args freely

Reply:
> This is quite confusing logic for someone new to this (especially since there is very little documentation).

Yeah it can be confusing. I am not a user of this feature/requirement so don't have any comments on the use case.
Regardless i leave it for your team, i personally feel that configuration management is a total abstraction, operator CR should be a minimal viable spec to support what is needed to support overrides, templating etc. :) 

Also @CodingParsley how about just adding StartScript to each nodeSpec also, currently is just present on the cluster scope ( its common to all nodesSpecs ). 
https://github.com/druid-io/druid-operator/blob/master/apis/druid/v1alpha1/druid_types.go#L95

Reply:
> just adding StartScript to each nodeSpec also, currently is just present on the cluster scope ( its common to all nodesSpecs )
Sounds like a solution! Will rewrite the PR